crates/bevy_ui/src/update.rs

```diff
@@ -92,3 +92,55 @@ fn update_clipping(
         }
     }
 }
+
+pub fn update_ui_camera_system(
+    mut commands: Commands,
+    root_node_query: Query<
+        (Entity, Option<&UiCamera>),
+        (With<Node>, Without<Parent>, Changed<UiCamera>),
+    >,
+    changed_children_query: Query<(Entity, &UiCamera), Changed<Children>>,
+    children_query: Query<&Children>,
+    node_query: Query<Option<&UiCamera>, With<Node>>,
+) {
+    for (root_node, camera) in &root_node_query {
+        update_ui_camera(
+            &mut commands,
+            &children_query,
+            &node_query,
+            root_node,
+            camera,
+        );
+    }
+
+    for (parent, camera) in &changed_children_query {
```

Comment:
This will do a lot of duplicated work when a deep tree is added. Would it be possible to avoid it?

Reply:
Also resolved by https://github.com/bevyengine/bevy/pull/10559/commits/0b45e1e8c07337dd441def4b02a5bf0388ae7949